When decoding, if JWT payload is not valid it returns null

dschenkelman opened the pull request.

hsablonniere commented:
Hey, BTW, there's already a "safe" JSON parse : https://github.com/brianloveswords/node-jws/blob/master/lib/verify-stream.js#L14

I have the same problem so I would really like to see this PR merged.

dschenkelman commented:
Wow, had forgotten about this PR! @hsablonniere thanks for the suggestion, updated the code.

Just updated to master in case this could still get in. I know it changes behaviors compared to the old one, but is seems more consistent.

Thoughts?

hsablonniere commented:
Hey Damian,

Long time no see :-)
I don't have any ongoing projects using this. I just took a bit of time to read it and it seems legit.

loucadufault commented on lib/verify-stream.js:
```diff
@@ -11,10 +11,10 @@ function isObject(thing) {
+function safeJsonParse(thing, encoding) {
   if (isObject(thing))
     return thing;
-  try { return JSON.parse(thing); }
```
this is not supported by the `JSON.parse` API, see comment here: https://github.com/auth0/node-jws/pull/86#discussion_r1090150451

Ref: `reviver` in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse

loucadufault replied:
IMO only the `if (!payload) { return null; }` check should be added by this CR, encoding support is addressed in newer CR #86
